src/cursor/abstract_cursor.ts

```diff
@@ -648,15 +620,29 @@ export abstract class AbstractCursor<
 
   /** @internal */
   async getMore(batchSize: number, useCursorResponse = false): Promise<Document | null> {
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const getMoreOperation = new GetMoreOperation(this[kNamespace], this[kId]!, this[kServer]!, {
-      ...this[kOptions],
-      session: this[kSession],
-      batchSize,
-      useCursorResponse
-    });
+    if (this.cursorId == null) {
+      throw new MongoRuntimeError(
+        'Unexpected null cursor id. A cursor creating command should have set this'
+      );
+    }
+    if (this.selectedServer == null) {
+      throw new MongoRuntimeError(
+        'Unexpected null selectedServer. A cursor creating command should have set this'
+      );
+    }
```

Comment:
Is this replacing the non-null assertion?

Reply:
Yes, happy to revert this, (or make sure we have coverage?). My thinking is that if this somehow was null and we pass it down the code below trusts it always exists but it doesn't necessarily mean the code below will use it in a way that's crashing, like dot access, it could go all the way to the server and we get "cursor.id cannot be null"